api/src/OpenTelemetry/Internal/Common/Types.hs

```diff
@@ -175,6 +200,25 @@ data FlushResult
   deriving (Show)
```

Comment:
I'd recommend tweaking `FlushError` a bit here to carry exceptions with it:

``` haskell
| FlushError [SomeException]
```

Reply:
Will do.